yaml/_dumper.ts

```diff
@@ -523,77 +523,63 @@ function blockHeader(string: string, indentPerLevel: number): string {
 //    • No ending newline => unaffected; already using strip "-" chomping.
 //    • Ending newline    => removed then restored.
 //  Importantly, this keeps the "+" chomp indicator from gaining an extra line.
-function writeScalar(
```

Comment:
These functions are consistently named `write<Thing>`. I don't think it's a good idea to break that consistency

Reply:
Also this change breaks the practice to assign to .dump field in each `write*` function. I'm not in favor of breaking that either.

Reply:
Yes, I am aware of that. I think we might want to change more with other prs, because `state.dump` is not very well defined. The type changes in different functions, with type any it is hard to follow and debug.

Reply:
> The type changes in different functions, with type any it is hard to follow and debug.

I think that is the reason to not change that until we write more robust testing because it's too risky to change, and also too difficult to validate that change is not breaking. `_dumper.ts` currently has 262 lines uncovered by testing. Can you first contribute to improve the test coverage?